src/components/stretchable-link/stretchable-link.tsx

```diff
@@ -0,0 +1,27 @@
+import classNames from 'classnames';
+import styles from './stretchable-link.module.css';
+import { Link } from '@remix-run/react';
+import { ROUTES } from '~/router/config';
+
+interface StretchableLinkProps {
+    title: string;
+    categorySlug: string;
+    className?: string;
+    children?: React.ReactNode;
+}
+
+export const StretchableLink = ({ title, className, categorySlug }: StretchableLinkProps) => {
```

Comment:
Doubting about the name. User can change icon and animation... Then it is just a link.